src/flask/app.py

```diff
@@ -1783,8 +1783,13 @@ def inject_url_defaults(self, endpoint: str, values: dict) -> None:
         """
         funcs: t.Iterable[URLDefaultCallable] = self.url_default_functions[None]
         if "." in endpoint:
-            bp = endpoint.rsplit(".", 1)[0]
-            funcs = chain(funcs, self.url_default_functions[bp])
+            bps: t.List[str] = [endpoint.rsplit(".", 1)[0]]
```

Comment:
Why doesn't this use `request.blueprints`?

Reply:
As the endpoint is directly available, and I'm not sure if we guarantee the request is available at all occasions when this can be called.